src/tokenizer.java

```diff
@@ -21,152 +21,115 @@
      */
     public static List<token> tokenize(String filepath) throws FileNotFoundException {
         Scanner sc = new Scanner(new File(filepath));
-        int lineCount = 1;  //An incrementing line counter
+        int lineCount = 0;  //An incrementing line counter
         List<token> tokenList = new LinkedList<>(); //The list of tokens which will be returned
 
         while (sc.hasNextLine()) {  //For each line of the source file
+            lineCount++;
             String line = sc.nextLine();    //Grab the line
             int lineLength = line.length(); //Grab its length once
 
-            for (int i = 0; i < lineLength; i++) {
-                //Repeat these actions while there are still chars on the line
+            for (int i = 0; i < lineLength; i++) { // Process every token on the line
                 StringBuilder tok = new StringBuilder();
-
-                //Clear whitespace to start of new token
-                while (i < lineLength) {
-                    char ch = line.charAt(i);
-                    if (!Character.isWhitespace(ch))
-                        break;
-                    i++;
-                }
-
-                while (i < lineLength) {    //Tokens end at the end of a line
-                    char ch = line.charAt(i);
-                    if (Character.isWhitespace(ch))  //Tokens also end on whitespace
-                        break;
-
-                    if (ch == ';' || ch == '(' || ch == ')' || ch == ',' || ch == '"') {
-                        if (!tok.toString().isEmpty()) {
+                char ch = line.charAt(i);
+                if (Character.isWhitespace(ch)) continue; // Skip whitespace before grabbing a new token
+                if (ch == '/' && ++i < lineLength && line.charAt(i) == '/') break; // Ignore comment
+                if (Character.isAlphabetic(ch)) { // Identifier or keyword
+                    while (i < lineLength) {
+                        ch = line.charAt(i);
+                        if (!Character.isAlphabetic(ch) && !Character.isDigit(ch)) {
                             i--;
                             break;
                         }
-                        else {
-                            tok.append(ch);
+                        tok.append(ch);
+                        i++;
+                    }
+                    // Check reserved keywords
+                    switch (tok.toString()) {
+                        case "print":
+                            tokenList.add(new print_label(lineCount));
```

Comment:
while we're here, Johnson mentioned in class yesterday that he meant for us to throw the position in the line as well as the line number. he said the writeup that demonstrated otherwise was bad but he changed it. I'm pretty sure he said he wasn't going to penalize in this phase if we just do line number, but he would for phase 2 so he said we should change it eventually.